src/libostree/ostree-sysroot-deploy.c

```diff
@@ -4270,6 +4261,120 @@ ostree_sysroot_deployment_set_mutable (OstreeSysroot *self, OstreeDeployment *de
   return TRUE;
 }
 
+/**
+ * ostree_sysroot_deployment_prepare_next_root
+ * @self: Sysroot
+ * @deployment: Deployment to prepare /run/nextroot for
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Prepare the specified deployment for a systemd soft-reboot by creating a new
+ * root with it at /run/nextroot
+ *
+ * Since: TODO
+ */
+gboolean
+ostree_sysroot_deployment_prepare_next_root (OstreeSysroot *self, OstreeDeployment *deployment,
+                                             GCancellable *cancellable, GError **error)
+{
+  GLNX_AUTO_PREFIX_ERROR ("Preparing /run/nextroot for a soft-reboot", error);
+  g_autofree char *deployment_path = ostree_sysroot_get_deployment_dirpath (self, deployment);
+  gint estatus;
+
+  glnx_autofd int deployment_dfd = -1;
+  if (!glnx_opendirat (self->sysroot_fd, deployment_path, FALSE, &deployment_dfd, error))
+    return FALSE;
+
+  const char *argv[] = { "/usr/lib/ostree/ostree-prepare-root", "--mount", "/run/nextroot",
+                         // we need the fd here instead of the path.
+                         deployment_path, NULL };
```

Comment:
I forgot removing this... but I also forgot if passing _ostree_sysroot_child_setup_fchdir was enough in the call.